src/main/java/io/curity/identityserver/plugin/twitter/authentication/TwitterAuthenticatorRequestHandler.java

```diff
@@ -66,21 +66,22 @@ public TwitterAuthenticatorRequestHandler(TwitterAuthenticatorPluginConfig confi
                 .apiSecret(_config.getClientSecret())
                 .callback(createRedirectUri())
                 .build(TwitterApi.instance());
-        String authorizationEndpoint = "";
-        
+
+
+        final OAuth1RequestToken requestToken;
         try
         {
-            final OAuth1RequestToken requestToken = service.getRequestToken();
-            authorizationEndpoint = service.getAuthorizationUrl(requestToken);
-            _config.getSessionManager().put(Attribute.of(OAUTH_TOKEN, requestToken.getToken()));
-            _config.getSessionManager().put(Attribute.of(OAUTH_TOKEN_SECRET, requestToken.getTokenSecret()));
-
+            requestToken = service.getRequestToken();
         }
         catch (Exception e)
         {
-            e.printStackTrace();
+            throw _exceptionFactory.externalServiceException("Error authenticating with Twitter: " + e);
```

Comment:
The stack trace is utterly uninteresting, so I just toString() it so that we can see what Twitter is complaining about.